elastic/kibana
# [Alerting] Add telemetry for query/search durations during rule execution

Resolves https://github.com/elastic/kibana/issues/125967

## Summary

Updates telemetry aggregation to query for ES search duration and total search duration. Calculates the average of these values and the average as broken down by rule type.

## To Verify

1. Update the alerting telemetry task to run more frequently:

```
--- a/x-pack/plugins/alerting/server/usage/task.ts
+++ b/x-pack/plugins/alerting/server/usage/task.ts
@@ -139,7 +139,7 @@ export function telemetryTaskRunner(
                   avg_execution_time_per_day: dailyExecutionCounts.avgExecutionTime,
                   avg_execution_time_by_type_per_day: dailyExecutionCounts.avgExecutionTimeByType,
                 },
-                runAt: getNextMidnight(),
+                runAt: new Date(), // getNextMidnight(),
               };
             }
           )
```

2. Create some rules that query ES (index threshold or Elasticsearch query both do) and let them run for a bit
3. Navigate to `https://localhost:5601/api/stats?extended=true&legacy=true` and see the following fields inside alerting telemetry:
* `alerts.avg_es_search_duration_per_day `
* `alerts.avg_es_search_duration_by_type_per_day `
* `alerts.avg_total_search_duration_per_day `
* `alerts.avg_total_search_duration_by_type_per_day`


### Checklist
